src/Lens.js

```diff
@@ -62,3 +63,18 @@ export const prop = (key) => lens(get(key), set(key))
 
 // ix : Number → Lens s a
 export const ix = (index) => lens(get(index), set(index))
+
+// alter : String → Lens (Maybe s) (Maybe a)
+export const alter = (key) =>
+  lens(
+    (obj) => (isNotFound(obj) ? notFound : obj[key] || notFound),
+    (val, obj) => {
+      if (isNotFound(val)) {
+        var newObj = { ...obj }
+        delete newObj[key]
```

Comment:
You can skip this `delete` by selectively destructuring the props you want to keep for the new object! 😉

Reply:
I don't know how to do that. Could you give me a pointer/hint? :)

Reply:
Fixed in 8fe59ab